File: iterableapi/src/main/java/com/iterable/iterableapi/IterablePushActionReceiver.java
package com.iterable.iterableapi;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Handles incoming push actions built by {@link IterableNotificationBuilder}
 * Action id is passed in the Intent extras under {@link IterableConstants#REQUEST_CODE}
 */
public class IterablePushActionReceiver extends BroadcastReceiver {
    private static final String TAG = "IterablePushActionReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        IterablePushNotificationUtil.dismissNotification(context, intent);
        IterablePushNotificationUtil.dismissNotificationPanel(context);
        String actionName = intent.getAction();
        if (IterableConstants.ACTION_PUSH_ACTION.equalsIgnoreCase(actionName)) {
            IterablePushNotificationUtil.handlePushAction(context, intent);
        }
    }
}

File: iterableapi/src/main/java/com/iterable/iterableapi/IterablePushActionReceiver.kt
package com.iterable.iterableapi

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

/**
 * Handles incoming push actions built by [IterableNotificationBuilder]
 * Action id is passed in the Intent extras under [IterableConstants.REQUEST_CODE]
 */
class IterablePushActionReceiver : BroadcastReceiver() {
    
    companion object {
        private const val TAG = "IterablePushActionReceiver"
    }

    override fun onReceive(context: Context, intent: Intent) {
        IterablePushNotificationUtil.dismissNotification(context, intent)
        IterablePushNotificationUtil.dismissNotificationPanel(context)
        val actionName = intent.action
        if (IterableConstants.ACTION_PUSH_ACTION.equals(actionName, ignoreCase = true)) {
            IterablePushNotificationUtil.handlePushAction(context, intent)
        }
    }
}
